src/tempfile.rs

```diff
@@ -51,7 +52,57 @@ impl TempFile {
     /// `prefix`: The path and filename where to create the temporary file. Six
     /// random alphanumeric characters will be added to the end of this to form
     /// the filename.
+    #[cfg(unix)]
     pub fn new_with_prefix<P: AsRef<OsStr>>(prefix: P) -> Result<TempFile> {
+        use std::ffi::CString;
+        use std::os::unix::{ffi::OsStrExt, io::FromRawFd};
+
+        let mut os_fname = prefix.as_ref().to_os_string();
+        os_fname.push("XXXXXX");
+
+        // When trying to open a file whose path has nul characters in it using rust's
+        // stdlib, the `From<io::Error> for Error` impl would result in `Error::new(0)`
+        let raw_fname = match CString::new(os_fname.as_bytes()) {
+            Ok(c_string) => c_string.into_raw(),
+            Err(_) => return Err(Error::new(0)),
```

Comment:
To keep consistency with the other modules exported by vmm-sys-util, could you please EINVAL as follows:
```
Error::new(libc::EINVAL)
```

This also makes the code explicit in terms of the error being returned.

Reply:
Done!